flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/result/ResultStore.java

```diff
@@ -37,6 +37,13 @@ public class ResultStore {
 
     private static final Logger LOG = LoggerFactory.getLogger(ResultStore.class);
 
+    public static final ResultStore DUMMY_RESULT_STORE =
+            new ResultStore(CloseableIterator.adapterForIterator(Collections.emptyIterator()), 0);
+
+    static {
+        DUMMY_RESULT_STORE.close();
```

Comment:
Because we don't want users to fetch results from the DUMMY store. When fetching from a closed store, the fetcher will know all results have been returned.